draft-ietf-sidrops-aspa-verification.xml

```diff
@@ -448,227 +348,151 @@ hop(AS(i), AS(j)) =  / Else, "Provider+" if the VAP-SPAS entry
+        The maximum up-ramp length can be defined as I, where I is the minimum index for which authorized(A(I), A(I+1)) returns 'Not Provider'. If there is no such I, the maximum is equal to AS_PATH length, or N. We define this parameter as max_up_ramp. The minimum up-ramp length can be defined as I, where I is minimum index for which authorized(A(I), A(I+1)) returns "No Attestation" or "Not Provider". If there is no such I, the AS_APATH consists only of "Provider+" pairs only, so the minimum up-ramp length equals to AS_PATH length, or N. We define this parameter as min_up_ramp.
+      </t>
+      <t>
+        Similar the maximum down-ramp length can be defined as N - J + 1 where J is maximum index for which authorized(A(J), A(J-1)) returns "Not Provider". If there is no such J, the maximum is equal to AS_PATH length, or N. We define this parameter as max_down_ramp. The minimum down-ramp length can be defined as N - J + 1 where J is maximum index for which authorized(A(J), A(J-1)) returns "No Attestation" or "Not Provider". If there is no such J, the minimum equals to AS_PATH length, or N. We define this parameter as min_down_ramp.
+      </t>
+
+      <t>
+        If sum of maximum estimations of up-ramp and down-ramp is less than N, the path is Invalid.
+        If sum of minimum estimations of up-ramp and down-ramp is less than N, the path is Unknown. Otherwise, it is Valid.
+      </t>
+
+      <t>
+        Below are formal procedures for path verification depending on the role between receiving AS and its neighbour.
+        These procedures are using compressed sequence representation of AS_PATH {AS(N), AS(N-1),..., AS(2), AS(1)} and defined with above parameters max_up_ramp, min_up_ramp, max_down_ramp and min_up_ramp
       </t>
     </section>
-    <section title="Formal Procedure for Verification of Downstream Paths" anchor="down-procedure">
+
+    <section title="Algorithm for Upstream Paths" anchor="Upflow">
       <t>
-        The downstream path verification procedure is formally specified as follows:
+        The upstream verification algorithm described here is applied when a route is received from a customer or lateral peer, or is received by an RS from an RS-client, or is received by an RS-client from an RS.
+        In all these cases, the receiving/validating eBGP router expects the AS_PATH does not have a down-ramp, only up-ramp.
+        Thus max_down_ramp and min_up_ramp are set to 0.
+      </t>
+      <t>
+        The upstream path verification procedure is specified as follows:
       </t>
       <t>
         <list style="numbers">
           <t>
             If the AS_PATH has an AS_SET, then the procedure halts with the outcome "Invalid".
           </t>
           <t>
-            Collapse prepends in the AS_SEQUENCE(s) in the AS_PATH (i.e., keep only the unique AS numbers).
```

Comment:
why did you remove this?

Reply:
I moved the disclaimer into the principles section:

>    Below are formal procedures for path verification depending on the
>    role between receiving AS and its neighbour.  These procedures are
>    using compressed sequence representation of AS_PATH {AS(N),
>    AS(N-1),..., AS(2), AS(1)} and defined with above parameters
>    max_up_ramp, min_up_ramp, max_down_ramp and min_up_ramp

As we don't redefine ramp variables, I don't see the reason to redefine compressed AS_PATH in the verification procedures.